### PR TITLE
Resolves issue #1 added filter query support

### PR DESCRIPTION
**Pull request by RushikeshSawant**
This PR resolves issue #1

### CODE REVIEW

File: src/main/java/org/leeu1911/JsonBoxObject.java [hunk 1]
```diff
@@ -30,6 +40,41 @@
...
+    public static <T> List<T> findByFilter(String filterQuery, Class<T> clazz) {
+        Objects.requireNonNull(filterQuery);
+        if(validFilterQuery(filterQuery)) {
+            filterQuery = encode(filterQuery);
+            HttpResponse response = get(filterQuery);
+            if (response.getStatusCode() > 299) {
+                return null;
+            }
+            Type type = TypeToken.getParameterized(ArrayList.class, clazz).getType();
+            return GSON.fromJson(response.getResponseBody(), type);
```
Comment (LeeU1911) [inline src/main/java/org/leeu1911/JsonBoxObject.java]: Please use `parseHttpResponse` method for this.

File: src/test/java/org/leeu1911/JsonBoxTest.groovy [hunk 2]
```diff
@@ -4,7 +4,7 @@ import spock.lang.Specification
 
 def class JsonBoxTest extends Specification {
     def setup() {
-        JsonBox.boxId = "json_box_java_test_22092019"
+        JsonBox.boxId = UUID.randomUUID().toString().replaceAll("-","_")
```
Comment (LeeU1911) [inline src/test/java/org/leeu1911/JsonBoxTest.groovy]: I wanted to fixed the `boxId` to not pollute jsonbox with test data for every time test suite runs. Did you have a problem with your tests so you had to change this? If not, let's revert this change.

**Review by LeeU1911 — COMMENTED** [review]
Thanks for your contribution. Looks good. Just a small comment or two then I can merge.

**Review by LeeU1911 — CHANGES_REQUESTED** [review]
Please update your PR with points mentioned.

**Comment by RushikeshSawant** [comment]
@LeeU1911 updated my PR to address your comments

**Pull request closed without merging**